Src/Common/AppInsightsActivity.cs

```diff
@@ -0,0 +1,145 @@
+﻿namespace Microsoft.ApplicationInsights.Common
+{
+    using System;
+    using System.Diagnostics;
+    using System.Threading;
+
+    // this is a temporary solution that mimics System.Diagnostics.Activity and Correlation HTTP protocol:
```

Comment:
nitpick: Thank-you for all the comments, I love comments! :)  Please convert them to XML doc comments.